cfnstack/assets.go

```diff
@@ -117,23 +119,33 @@ type Asset struct {
 
 type AssetLocationProvider struct {
 	s3URI     string
+	s3Region  string
 	stackName string
 }
 
 type AssetLocation struct {
-	ID     AssetID
-	Key    string
-	Bucket string
-	Path   string
+	ID       AssetID
+	Key      string
+	Bucket   string
+	Path     string
+	S3Region string
 }
 
 func (l AssetLocation) URL() string {
-	return fmt.Sprintf("https://s3.amazonaws.com/%s/%s", l.Bucket, l.Key)
+	var url string
+	if strings.HasPrefix(l.S3Region, "cn") {
+		url = fmt.Sprintf("https://s3.cn-north-1.amazonaws.com.cn/%s/%s", l.Bucket, l.Key)
+	} else {
+		url = fmt.Sprintf("https://s3.amazonaws.com/%s/%s", l.Bucket, l.Key)
+	}
+	log.Printf("Using S3 URL: %s on region: %s", url, l.S3Region)
+	return url
```

Comment:
As we have the `Region` type in `model/region.go` today, perhaps we'd better implement `func (r Region) S3EndpointName() string` to return e.g. `"s3.cn-north-1.amazonaws.com.cn"` for `cn-north-1` or `"s3.amazonaws.com"` otherwise, to be used here.